internal/observability/middleware.go

```diff
@@ -152,14 +163,26 @@ func (m *Middleware) Handle(ctx context.Context, req *transport.Request, w trans
 	call := m.graph.begin(ctx, transport.Unary, _directionInbound, req)
 	defer m.handlePanicForCall(call, transport.Unary)
 
+	var requestSize int
+
+	if call.direction == _directionInbound {
+		if wrapper, ok := req.Body.(lenWrapper); ok {
```

Comment:
Let's add some comments here about why we have decided to do this like this and the drawbacks (save one copy but if a middleware of an application change the body with something which does not inplement lenWrapper - no stats)